tests/nodetree/test_node_unit.py

```diff
@@ -813,23 +851,317 @@ async def test_wait_for_state_change_loop(
         nr_expected_calls,
     ):
         node = MockLeafNode(())
-        queue = Mock()
-        queue.get = Mock(
-            side_effect=[
-                self._get_future(AnnotatedValue(value=i, path="/")) for i in in_pipe
-            ],
-        )
+        queue = create_autospec(DataQueue)
+        queue.get.side_effect = [AnnotatedValue(value=i, path="/") for i in in_pipe]
+
         await node._wait_for_state_change_loop(queue, target_value, invert=invert)
         queue.get.call_count = nr_expected_calls
 
-    def test_node_info(self, mock_path):
+    def test_node_info(self, mock_path, zi_structure):
         node = MockLeafNode(())
-        node._tree_manager = Mock()
+        node._tree_manager = create_autospec(NodeTreeManager)
         node._tree_manager.path_to_info = {"path": "this_info"}
-        with patch(
-            "labone.nodetree.node.NodeInfo.__init__",
+
+        info = zi_structure.nodes_to_info["/zi/debug/level"]
+        node_info = NodeInfo(info)
+
+        def fake_init_func(self_, *_, **__):
+            self_._info = info
+
+        with patch.object(
+            NodeInfo,
+            "__init__",
             return_value=None,
+            side_effect=fake_init_func,
+            autospec=True,
         ) as node_info_mock:
             node.node_info  # noqa: B018
             mock_path.assert_called_once()
-            node_info_mock.assert_called_once_with("this_info")
+            node_info_mock.assert_called_once_with(node_info, "this_info")
+
+
+class TestWildCardOrPartialNode:
+    @pytest.mark.asyncio()
+    async def test_get(self, mock_path):
+        node = MockWildcardOrPartialNode(())
+        node._package_get_response = Mock(return_value="package_get_response")
+        node._tree_manager = create_autospec(NodeTreeManager)
+        node._tree_manager._session = create_autospec(Session)
+        node.tree_manager.session.get_with_expression.return_value = _get_future(
+            "get_response",
+        )
+
+        result = await node._get()
+        mock_path.assert_called_once()
+        node.tree_manager.session.get_with_expression.assert_called_once_with("path")
+        node._package_get_response.assert_called_once_with("get_response")
+        assert result == "package_get_response"
+
+    @pytest.mark.asyncio()
+    async def test_set(self, mock_path):
+        node = MockWildcardOrPartialNode(())
+        node._package_get_response = Mock(
+            return_value="package_get_response", spec=True,
```

Comment:
nit-pick: `package_set_response`

Reply:
I understand the confusion, but actually, we only have a _package_get_response function. It's purpose is to be used both for the return values of get and set, since they are similar. Is it worth renaming to something like _package_response?

Reply:
Its still only a string you tell a mock to return so I would adapt the string to `package_set_response`